lib/modules/platform/bitbucket-server/index.ts

```diff
@@ -1154,6 +1156,152 @@ export async function mergePr({
   return true;
 }
 
+export async function expandGroupMembers(
+  reviewers: string[],
+): Promise<string[]> {
+  logger.debug(`expandGroupMembers(${reviewers.join(', ')})`);
+  const expandedUsers: string[] = [];
+
+  for (const reviewer of reviewers) {
+    const [baseEntry, modifier] = reviewer.split(':');
+
+    if (isReviewerGroup(baseEntry)) {
+      const groupName = baseEntry.replace(/^@reviewer-group\//, '');
+      const groupUsers = await getUsersFromReviewerGroup(groupName);
+      if (groupUsers.length === 0) {
+        continue;
+      }
+
+      if (modifier === undefined) {
+        expandedUsers.push(...groupUsers); // Add all users from the group
+        continue;
+      }
+      const randomCount = parseModifier(modifier);
+      if (randomCount === null) {
+        expandedUsers.push(...groupUsers);
+        continue;
+      }
+      expandedUsers.push(...resolveRandom(groupUsers, randomCount));
+    } else {
+      expandedUsers.push(baseEntry); // Add the user entry
+    }
+  }
+
+  return [...new Set(expandedUsers)];
+}
+
+export function extractRulesFromCodeOwnersLines(
+  cleanedLines: string[],
+): FileOwnerRule[] {
+  const results: FileOwnerRule[] = [];
+
+  const reversedLines = cleanedLines
+    .filter((line) => line.trim() !== '' && !line.trim().startsWith('#'))
+    .reverse();
+
+  for (const line of reversedLines) {
+    const safeLine = line.replace(/\\\\ /g, '<<SPACE>>'); // double-backslash before space
+    const [pattern, ...rawEntries] = safeLine.trim().split(/\s+/);
+    const realEntries = rawEntries.map((u) => u.replace(/<<SPACE>>/g, ' '));
+
+    const matcher = ignore().add(pattern);
+    results.push({
+      pattern,
+      usernames: [...new Set(realEntries)],
+      score: pattern.length,
+      match: (path: string) => matcher.ignores(path),
+    });
+  }
+
+  return results;
+}
+
+function isReviewerGroup(entry: string): boolean {
+  return entry.startsWith('@reviewer-group/');
+}
+
+function parseModifier(value: string): number | null {
+  const randomPrefix = 'random';
+
+  if (value.startsWith(randomPrefix)) {
+    const match = /^random\((\d+)\)$/.exec(value);
+    if (match) {
+      return parseInt(match[1], 10); // e.g., random(2) → 2
+    } else {
+      return 1; // plain "random"
+    }
+  }
+
+  return null; // Not a random selector
```

Comment:
This could be simplified to:
```suggestion
  const match = /^random(?:\((\d+)\))?$/.exec(value);
  if (match) {
    return parseInt(match[1] ?? "1", 10); // e.g., random(2) → 2
  } else {
    return null;
  }
```

There should be a test case for the `return null` scenario too